### PR TITLE
[Doc] add auto_increment

### PR DESCRIPTION
**Pull request by hellolilyliuyi**
## What type of PR is this：
- [ ] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [x] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes #

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto backported to target branch
  - [x] 3.0
  - [ ] 2.5
  - [ ] 2.4
  - [ ] 2.3

### CODE REVIEW

File: docs/sql-reference/sql-statements/auto_increment.md [hunk 1]
```diff
@@ -0,0 +1,328 @@
...
+
+- Serve as primary keys:  `AUTO_INCREMENT` columns can be used as primary keys to ensure that each record has a unique ID and make it easy to query and manage data.
+- Join tables: When multiple tables are joined, `AUTO_INCREMENT` columns can be used as Join Keys, which can expedite queries compared to using columns whose data types are UUID and STRING.
+- Count the number of distinct values in a high-cardinality column: An `AUTO_INCREMENT` column can be used to represent the unique value column in a dictionary. Compared to directly counting distinct STRING values, counting distinct integer values of the`AUTO_INCREMENT` column can sometimes improve the query speed by several times or even tens of times.
+
+You need to specify `AUTO_INCREMENT` columns in the CREATE TABLE statement. The data types of the `AUTO_INCREMENT` columns' must be BIGINT. The value for a `AUTO_INCREMENT` column starts from 1, and increments by 1 for each new record. The values for the `AUTO_INCREMENT` column can be [implicitly assigned](#Assign values implicitly) or [explicitly specified](#Specify values explicitly).
```
Comment (amber-create) [inline docs/sql-reference/sql-statements/auto_increment.md:13]: image
冠词 a 改为 -> an

Comment (amber-create) [inline docs/sql-reference/sql-statements/auto_increment.md:13]: image
非特指，删掉 the

Comment (amber-create) [inline docs/sql-reference/sql-statements/auto_increment.md:13]: image
改为 -> The values for an `AUTO_INCREMENT` column can be [implicitly assigned](#Assign values implicitly) or [explicitly specified](#Specify values explicitly). They start from 1, and increment by 1 for each new record.

File: docs/sql-reference/sql-statements/auto_increment.md [hunk 2]
```diff
@@ -0,0 +1,328 @@
...
+
+Explicitly assign and implicitly specify the values for the `AUTO_INCREMENT` column `number` in the table `t`。
+
+```SQL
+INSERT INTO t (id, number) VALUES (1, DEFAULT);
+INSERT INTO t (id, number) VALUES (2, 2);
+INSERT INTO t (id, number) VALUES (3);
+```
+
+View the table `t`.
```
Comment (amber-create) [inline docs/sql-reference/sql-statements/auto_increment.md:119]: 一般说 Query the table

File: docs/sql-reference/sql-statements/auto_increment.md [hunk 3]
```diff
@@ -0,0 +1,328 @@
...
+|    5 |      3 |
++------+--------+
+5 rows in set (0.07 sec)
+```
+
+## Partial updates and `AUTO_INCREMENT` columns
+
+This section explains how to update only a few specified columns in a table that contains an `AUTO_INCREMENT` column.
+
+> **NOTE** Currently, only primary key tables support partial updates.
```
Comment (amber-create) [inline docs/sql-reference/sql-statements/auto_increment.md:180]: image
Currently 那句要换行

File: docs/sql-reference/sql-statements/auto_increment.md [hunk 4]
```diff
@@ -0,0 +1,328 @@
...
+    PRIMARY KEY (id, name)
+    DISTRIBUTED BY HASH(id)
+    PROPERTIES("replicated_storage" = "true");
+
+    -- prepare data
+    INSERT INTO t (id, name, job1, job2) VALUES (0, 0, 1, 1);
+    Query OK, 1 row affected (0.04 sec)
+    {'label':'insert_6af28e77-7d2b-11ed-af6e-02424283676b', 'status':'VISIBLE', 'txnId':'152'}
+
+    -- view the table
```
Comment (amber-create) [inline docs/sql-reference/sql-statements/auto_increment.md:206]: view -> query

File: docs/sql-reference/sql-statements/auto_increment.md [hunk 5]
```diff
@@ -0,0 +1,328 @@
...
+    PRIMARY KEY (id, name)
+    DISTRIBUTED BY HASH(id)
+    PROPERTIES("replicated_storage" = "true");
+
+    -- Prepare data.
+    INSERT INTO t VALUES (0, 0, -1, -1);
+    Query OK, 1 row affected (0.04 sec)
+    {'label':'insert_458d9487-80f6-11ed-ae56-aa528ccd0ebf', 'status':'VISIBLE', 'txnId':'94'}
+
+    -- View the table.
```
Comment (amber-create) [inline docs/sql-reference/sql-statements/auto_increment.md:276]: View -> Query

File: docs/sql-reference/sql-statements/auto_increment.md [hunk 6]
```diff
@@ -0,0 +1,328 @@
...
+> **NOTE** Currently, only primary key tables support partial updates.
+
+### `AUTO_INCREMENT` column is primary key
+
+You need to specify the primary key during partial updates. Therefore, if the `AUTO_INCREMENT` column is the primary key or part of the primary key, the user behavior for partial updates is exactly the same as when the `AUTO_INCREMENT` column is not defined.
+
+1. Create a table `t` in the database `example_db` and insert one data row.
+
+    ```SQL
+    -- create a table
```
Comment (amber-create) [inline docs/sql-reference/sql-statements/auto_increment.md:189]: Create a table.

File: docs/sql-reference/sql-statements/auto_increment.md [hunk 7]
```diff
@@ -0,0 +1,328 @@
...
+        id BIGINT AUTO_INCREMENT,
+        name BIGINT NOT NULL,
+        job1 BIGINT NOT NULL,
+        job2 BIGINT NOT NULL
+    ) 
+    PRIMARY KEY (id, name)
+    DISTRIBUTED BY HASH(id)
+    PROPERTIES("replicated_storage" = "true");
+
+    -- prepare data
```
Comment (amber-create) [inline docs/sql-reference/sql-statements/auto_increment.md:201]: Prepared data.

File: docs/sql-reference/sql-statements/auto_increment.md [hunk 8]
```diff
@@ -0,0 +1,328 @@
...
+    ```Bash
+    curl --location-trusted -u root: -H "label:123" \
+        -H "column_separator:," \
+        -H "partial_update:true" \
+        -H "columns: id,name,job2" \
+        -T my_data.csv -XPUT \
+        http://<fe_host>:<fe_http_port>/api/example_db/t/_stream_load
+    ```
+
+4. View the updated table. The first row of data already exists in table `t`, so the `AUTO_INCREMENT` column `job1` retains its original value. The second and third rows of data are newly inserted, so StarRocks generate new values for the `AUTO_INCREMENT` column `job1`.
```
Comment (amber-create) [inline docs/sql-reference/sql-statements/auto_increment.md:305]: Query xxx

File: docs/sql-reference/sql-statements/auto_increment.md [hunk 9]
```diff
@@ -0,0 +1,328 @@
...
+
+## Limits
+
+- When a table with an `AUTO_INCREMENT` column is created, `"replicated_storage" = "true"` must be set to ensure that all replicas have the same auto-incremented IDs.
+- Each table can have only one `AUTO_INCREMENT` column.
+- The data type of the `AUTO_INCREMENT` column must be BIGINT.
+- The `AUTO_INCREMENT` column must be `NOT NULL` and does not have a default value.
+- If the `AUTO_INCREMENT` column is the primary key or part of the primary key, you can delete data in the Primary Key table. Otherwise, you cannot delete data in the Primary Key table.
+- Adding the `AUTO_INCREMENT` attribute by using ALTER TABLE is not supported.
+- Currently, StarRocks's shared-data mode does not support this feature.
```
Comment (amber-create) [inline docs/sql-reference/sql-statements/auto_increment.md:327]: 前面没太提到 this feature，改为 -> the `AUTO_INCREMENT` attribute 吧

File: docs/sql-reference/sql-statements/data-definition/CREATE TABLE.md [hunk 10]
```diff
@@ -80,6 +80,8 @@ This aggregation type applies ONLY to the aggregation model whose key_desc type
 - **DEFAULT <default_value>**: Use a given value of the column data type as the default value. For example, if the data type of the column is VARCHAR, you can specify a VARCHAR string, such as beijing, as the default value, as presented in `DEFAULT "beijing"`. Note that default values cannot be any of the following types: ARRAY, BITMAP, JSON, HLL, and BOOLEAN.
 - **DEFAULT (\<expr\>)**: Use the result returned by a given function as the default value. Only the [uuid()](../../sql-functions/utility-functions/uuid.md) and [uuid_numeric()](../../sql-functions/utility-functions/uuid_numeric.md) expressions are supported.
 
+- **AUTO_INCREMENT**：specifies an `AUTO_INCREMENT` column. The data types of the `AUTO_INCREMENT` columns' must be BIGINT. The auto-incremented ID starts from 1 and increases by 1. For more information about `AUTO_INCREMENT` columns, see [AUTO_INCREMENT](../../sql-statements/auto_increment.md).
```
Comment (amber-create) [inline docs/sql-reference/sql-statements/data-definition/CREATE TABLE.md:83]: image
非特指，the 删掉；columns 后面多了个单引号，删掉

Comment (amber-create) [inline docs/sql-reference/sql-statements/data-definition/CREATE TABLE.md:83]: image
改为-> Auto-incremented IDs start from 1 and increase at a step of 1.

**Review by amber-create — APPROVED** [review]

**Review by evelynzhaojie — APPROVED** [review]

**Comment by hellolilyliuyi** [comment]
@mergify backport branch-3.0

**Pull request merged**